File: mvp_compat_views/src/dagger/java/com/mandria/android/mvp/di/CoreModule.java
package com.mandria.android.mvp.di;

import com.mandria.android.mvp.PresenterCache;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Dagger module which will provide core application scoped objects.
 */
@Module
public class CoreModule {

    /**
     * Provides the presenter cache.
     *
     * @return Presenter cache.
     */
    @Singleton
    @Provides
    public PresenterCache getPresenterCache() {
        return new PresenterCache();
    }
}

File: views_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/di/CoreModule.kt
package com.mandria.android.mvp.di

import com.mandria.android.mvp.PresenterCache
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/** Dagger module which will provide core application scoped objects. **/
@Module
class CoreModule {

    /** Provides the presenter cache. **/
    @Provides
    @Singleton
    fun getPresenterCache(): PresenterCache = PresenterCache()
}